control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -71,64 +74,198 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	gwc := &gwv1beta1.GatewayClass{}
 	err = r.Client.Get(ctx, types.NamespacedName{Name: string(gw.Spec.GatewayClassName)}, gwc)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			return r.cleanupGatewayResources(ctx, log, gw)
```

Comment:
+1 I need this for my next PR

Reply:
No, so I dropped this because we have a check in the binder for the existence of the GatewayClass -- if the GatewayClass doesn't exist, then it hooks in and marks everything as needing to be cleaned up.